### PR TITLE
Add Romanian more fake datas.

### PR DESCRIPTION
**Pull request by feketemihai**
### What does this changes

Improve Romanian fake data generation, added more generators with Romanain datas

### CODE REVIEW

File: faker/providers/ssn/ro_RO/__init__.py [hunk 1]
```diff
@@ -25,3 +34,32 @@ def vat_id(self):
         """
 
         return self.bothify(self.random_element(self.vat_id_formats))
+
+    ssn_formats = ("#############",)
+
+    def ssn(self):
+        """
+        Mexican Social Security Number, as given by IMSS.
```
Comment (fcurella) [inline faker/providers/ssn/ro_RO/__init__.py:42]: You probably meant "Romanian"?

File: faker/providers/ssn/ro_RO/__init__.py [hunk 2]
```diff
@@ -1,6 +1,15 @@
 from .. import Provider as BaseProvider
 
 
+def ssn_checksum(number):
+    """
+    Calculate the checksum for the romanian SSN (CNP).
+    """
+    weights = (2, 7, 9, 1, 4, 6, 3, 5, 8, 2, 7, 9)
+    check = sum(w * int(n) for w, n in zip(weights, number)) % 11
+    return '1' if check == 10 else str(check)
```
Comment (fcurella) [inline faker/providers/ssn/ro_RO/__init__.py:10]: return an `int` instead of a `str`. You're coercing anyway in `ssn`.

**Review by fcurella — CHANGES_REQUESTED** [review]
Can you add tests for the new providers you're adding?

**Comment by feketemihai** [comment]
@fcurella I added the unit test for Romanian new stuff, but all tests are failing from :

=============================== warnings summary ===============================
faker/providers/person/fr_QC/__init__.py:8
tests/test_factory.py::FactoryTestCase::test_documentor
tests/test_providers_formats.py::test_no_invalid_formats[fr_QC]
  /home/runner/work/faker/faker/faker/providers/person/fr_QC/__init__.py:8: UserWarning: fr_QC locale is deprecated. Please use fr_CA.
    warnings.warn("fr_QC locale is deprecated. Please use fr_CA.")

tests/test_factory.py::FactoryTestCase::test_documentor
  /home/runner/work/faker/faker/faker/documentor.py:91: UserWarning: No appropriate country for country code AA
    warnings.warn(str(e))

tests/test_factory.py::FactoryTestCase::test_documentor
  /home/runner/work/faker/faker/faker/documentor.py:91: UserWarning: No appropriate country for country code QC
    warnings.warn(str(e))

tests/test_factory.py::FactoryTestCase::test_documentor
  /home/runner/work/faker/faker/faker/documentor.py:91: UserWarning: Country code cannot be determined from locale
    warnings.warn(str(e))

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED tests/providers/test_address.py::TestBaseProvider::test_administrative_unit_all_locales
======= 1 failed, 1208 passed, 8 skipped, 6 warnings in 78.60s (0:01:18) =======
ERROR: InvocationError for command /home/runner/work/faker/faker/.tox/py/bin/coverage run --source=faker -m pytest (exited with code 1)

**Comment by fcurella** [comment]
@feketemihai 

The issue is that Address providers now requires an `administrative_area` attribute, and the [new `ro_RO` provider is not defining one](https://github.com/joke2k/faker/pull/1389/checks?check_run_id=2047000933#step:5:1266)

File: faker/providers/address/ro_RO/__init__.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        return self.random_element(self.cities)
+
+    def city_with_postcode(self):
+        return self.postcode() + " " + self.random_element(self.cities)
+
+    def state(self):
+        """
+        example: u'Timiș'
+        """
+        return self.random_element(self.states)[1]
```
Comment (fcurella) [inline faker/providers/address/ro_RO/__init__.py:124]: can you rename this to `administrative_unit` and alias it to `state`?

```suggestion
    def administrative_unit(self):
        """
        example: u'Timiș'
        """
        return self.random_element(self.states)[1]
    
    state = administrative_unit
```

**Review by fcurella — CHANGES_REQUESTED** [review]

File: faker/providers/address/ro_RO/__init__.py (hunk 3 above)
Comment (feketemihai) [reply faker/providers/address/ro_RO/__init__.py]: Updated, thanks.

**Comment by fcurella** [comment]
Thank you!

**Pull request merged**